File: 907-KokoEatingBananas/907-KokoEatingBananas.py
# Last updated: 2/3/2026, 9:38:14 PM
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        low,high=1,max(piles)
        ans=0
        while low<=high:
            mid=low+(high-low)//2
            totalH=0
            for pile in piles:
                totalH+=math.ceil(pile/mid)
            if totalH<=h:
                ans=mid
                high =mid-1
            else:
                low=mid+1
        return ans


                



        

File: 907-KokoEatingBananas/907-KokoEatingBananas.java
// Last updated: 2/3/2026, 9:38:22 PM
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int l=1;
        int r=Arrays.stream(piles).max().getAsInt();
        int res=r;
        while(l<=r){
            int mid=(l+r)/2;
            int Time=0;
            for(int i :piles){
                Time+=Math.ceil((double)i/mid);
            }
            if (Time<=h){
                res=mid;
                r=mid-1;
            }
            else{
                l=mid+1;
            }
        }
        return res;
    }
}